velox/exec/Window.cpp

```diff
@@ -376,15 +548,38 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
 
     for (auto w = 0; w < numFuncs; w++) {
-      auto frameEndPoints = findFrameEndPoints(
-          w,
+      std::optional<vector_size_t> startOffset = std::nullopt;
+      std::optional<vector_size_t> endOffset = std::nullopt;
+      if (windowFrames_[w].startFrameArg) {
+        startOffset = constantPrecedingFrameOffset[w];
+      }
+      if (windowFrames_[w].endFrameArg) {
+        endOffset = constantFollowingFrameOffset[w];
+      }
+
+      rawFrameStartBuffers[w][j] =
+          windowFunctions_[w]->getFrameStartBoundFunction()(
+              windowFrames_[w].type,
+              w,
+              firstPartitionRow,
+              lastPartitionRow,
+              i,
+              startOffset) -
+          firstPartitionRow;
+      rawFrameEndBuffers[w][j] =
+          windowFunctions_[w]->getFrameEndBoundFunction()(
+              windowFrames_[w].type,
+              w,
+              firstPartitionRow,
+              lastPartitionRow,
+              i,
+              endOffset) -
+          firstPartitionRow;
+      emptyFrameCheck(
```

Comment:
Why are you invoking emptyFrameCheck at this point ? We logically cannot have empty frame is many of the frame combinations. Since emptyFrameCheck is dependent of particular frame combinations maybe you can use the function pointer approach here also.

Reply:
Thanks for the suggestion but I am not sure how the function pointer approach could be used here, could you please elaborate? I tried moving this to a class variable and removing this check but couldn't do so as the emptyFrameCheck is to be done after the rawFrameBuffers are computed for this index.